packages/tools/src/eventListeners/touch/touchStartListener.ts

```diff
@@ -290,126 +333,128 @@ function _onTouchDrag(evt: TouchEvent) {
-        ? getDeltaDistanceBetweenIPoints(currentPointsList, lastPointsList)
-        : getDeltaDistanceBetweenIPoints(currentPointsList, currentPointsList);
-    const eventDetail: EventTypes.TouchEndEventDetail = {
-      event: evt,
-      eventName: TOUCH_END,
-      element: state.element,
-      renderingEngineId: state.renderingEngineId,
-      viewportId: state.viewportId,
-      camera: {},
-      startPointsList: copyPointsList(state.startPointsList),
-      lastPointsList: copyPointsList(lastPointsList),
-      currentPointsList,
-      startPoints: getMeanTouchPoints(state.startPointsList),
-      lastPoints: getMeanTouchPoints(lastPointsList),
-      currentPoints: getMeanTouchPoints(currentPointsList),
-      deltaPoints,
-      deltaDistance,
-    };
-
-    triggerEventCallback(eventDetail.element, TOUCH_END, eventDetail);
-    state = JSON.parse(JSON.stringify(defaultState));
-
-    // Remove our temporary handlers which is only added when normalized touch
-    // start fires
-    document.removeEventListener('touchmove', _onTouchDrag);
-  } else {
-    // a tap occured
-    _onTouchTap(evt);
-  }
-
-  document.removeEventListener('touchend', _onTouchEnd);
-}
-
-function _onTouchTap(evt: TouchEvent): void {
   const currentPointsList = getTouchEventPoints(evt, state.element);
   const lastPointsList = _updateTouchEventsLastPoints(
     state.element,
     state.lastPointsList
   );
+  const deltaPoints =
+    currentPointsList.length === lastPointsList.length
+      ? getDeltaPoints(currentPointsList, lastPointsList)
+      : getDeltaPoints(currentPointsList, currentPointsList);
+  const deltaDistance =
+    currentPointsList.length === lastPointsList.length
+      ? getDeltaDistanceBetweenIPoints(currentPointsList, lastPointsList)
+      : getDeltaDistanceBetweenIPoints(currentPointsList, currentPointsList);
+  const eventDetail: EventTypes.TouchEndEventDetail = {
+    event: evt,
+    eventName: TOUCH_END,
+    element: state.element,
+    renderingEngineId: state.renderingEngineId,
+    viewportId: state.viewportId,
+    camera: {},
+    startPointsList: copyPointsList(state.startPointsList),
+    lastPointsList: copyPointsList(lastPointsList),
+    currentPointsList,
+    startPoints: getMeanTouchPoints(state.startPointsList),
+    lastPoints: getMeanTouchPoints(lastPointsList),
+    currentPoints: getMeanTouchPoints(currentPointsList),
+    deltaPoints,
+    deltaDistance,
+  };
+
+  triggerEventCallback(eventDetail.element, TOUCH_END, eventDetail);
+  _checkTouchTap(evt);
 
+  // reset to default state
+  state = JSON.parse(JSON.stringify(defaultState));
+  document.removeEventListener('touchmove', _onTouchDrag);
+  document.removeEventListener('touchend', _onTouchEnd);
+}
+
+function _checkTouchTap(evt: TouchEvent): void {
+  const currentTime = new Date().getTime();
+  const startTime = state.startTime.getTime();
+  if (currentTime - startTime >= tapState.tapWindow) return;
```

Comment:
I am a bit confused as to why two similar checks are made in this function: this line (379) and line 391. This one feels more correct than the other.

I think we want `>` here instead of `>=`.

Reply:
This is to check the current time against the touch_start start time. The other checks the currentTime against the tapState startTime.

Reply:
👍